specs/src/specs/header-sync.md

```diff
@@ -0,0 +1,47 @@
+# P2P Header Sync
+
+## Abstract
+
+The P2P Header Sync is a p2pP exchange service for rollkit headers that implements the [go-header](https://github.com/celestiaorg/go-header) interface. The main components are listed in the table below.
+
+| Component  | Description                                                                                                                                                                |
+| ---------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| store      | a `headerEx` prefixed datastore where synced headers are stored                                                                                                          |
+| subscriber | a libp2p node pubsub subscriber                                                                                                                                            |
+| p2p server | a server for handling header requests between peers in the p2p network                                                                                                     |
+| exchange   | a client that enables sending in/out-bound header requests from/to the p2p network                                                                                         |
+| syncer     | a service for efficient synchronization for headers. When a p2p node falls behind and wants to catch up to the latest network head via p2p network, it can use the syncer. |
+
+## Details
+
+All three types of nodes (sequencer, full, and light) run the P2P header exchange service to maintain the cannonical view of the rollup chain (with respect to the p2p network).
+
+The header exchange service inherits the ConnectionGater from the node's p2p client which allows blocking and allowing peers as needed by specifying the `P2PConfig.BlockedPeers` and `P2PConfig.AllowedPeers`.
```

Comment:
nit allows --> enables or controls to make it a little easier to read with allowing being what it allows. 

```suggestion
The header exchange service inherits the `ConnectionGater` from the node's p2p client which enables blocking and allowing peers as needed by specifying the `P2PConfig.BlockedPeers` and `P2PConfig.AllowedPeers`.
```